File: c++/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        size_t read = 0;
        size_t write = 0;

        while(read < nums.size()) {
            if(nums[read] != val) {
                if(write != read) {
                    nums[write] = nums[read];
                }
                write += 1;
            }

            read += 1;
        }

        return write;
    }
};

File: dev-rust/0027-remove-element.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut i_read  = 0;
        let mut i_write = 0;

        while i_read < nums.len() {

            if nums[i_read] != val {
                if i_read != i_write {
                    nums[i_write] = nums[i_read];
                } 

                i_write += 1;
            }

            i_read += 1;
        }

        return i_write as i32;
    }
}